### PR TITLE
fix `__time` column selector for incremental index

### PR DESCRIPTION
**Pull request by xvrl**
- also adds various tests for selecting the time column

### CODE REVIEW

File: processing/src/test/java/io/druid/query/timeseries/TimeseriesQueryRunnerTest.java [hunk 1]
```diff
@@ -2017,4 +2017,41 @@ public void testTimeSeriesWithFilteredAggInvertedNullValue()
...
+        )
+    );
+
+    Iterable<Result<TimeseriesResultValue>> actualResults = Sequences.toList(
+        runner.run(query, CONTEXT),
+        Lists.<Result<TimeseriesResultValue>>newArrayList()
+    );
+
+    TestHelper.assertExpectedResults(expectedResults, actualResults);
+  }
```
Comment (fjy) [inline processing/src/test/java/io/druid/query/timeseries/TimeseriesQueryRunnerTest.java]: can we add a test for topN as well?

File: processing/src/test/java/io/druid/query/QueryRunnerTestHelper.java [hunk 2]
```diff
@@ -92,19 +92,30 @@ public TableDataSource apply(@Nullable String input)
...
       Arrays.asList("placementish", "index"),
       "function aggregate(current, a, b) { if ((Array.isArray(a) && a.indexOf('a') > -1) || a === 'a') { return current + b; } else { return current; } }",
-      "function reset() { return 0; }",
-      "function combine(a, b) { return a + b; }"
+      JS_RESET_0,
+      JS_COMBINE_A_PLUS_B
+  );
+  public static final JavaScriptAggregatorFactory jsCountIfTimeGreaterThan = new JavaScriptAggregatorFactory(
+      "ntimestamps",
+      Arrays.asList("__time"),
```
Comment (fjy) [inline processing/src/test/java/io/druid/query/QueryRunnerTestHelper.java]: does casing matter here?

File: processing/src/test/java/io/druid/query/timeseries/TimeseriesQueryRunnerTest.java (hunk 1 above)
Comment (xvrl) [inline processing/src/test/java/io/druid/query/timeseries/TimeseriesQueryRunnerTest.java]: done

File: processing/src/test/java/io/druid/query/QueryRunnerTestHelper.java (hunk 2 above)
Comment (xvrl) [inline processing/src/test/java/io/druid/query/QueryRunnerTestHelper.java]: Druid is case sensitive now, so yes.

**Pull request merged**